server/src/main/java/org/elasticsearch/action/admin/indices/rollover/TransportRolloverAction.java

```diff
@@ -280,17 +282,13 @@ protected void masterOperation(
                     DataStream dataStream = (DataStream) indexAbstraction;
                     final Optional<IndexStats> indexStats = Optional.ofNullable(statsResponse)
                         .map(stats -> stats.getIndex(dataStream.getWriteIndex().getName()));
-
-                    Double indexWriteLoad = indexStats.map(
-                        stats -> Arrays.stream(stats.getShards())
-                            .filter(shardStats -> shardStats.getStats().indexing != null)
-                            // only take primaries into account as in stateful the replicas also index data
-                            .filter(shardStats -> shardStats.getShardRouting().primary())
-                            .map(shardStats -> shardStats.getStats().indexing.getTotal().getWriteLoad())
-                            .reduce(0.0, Double::sum)
-                    ).orElse(null);
-
-                    rolloverAutoSharding = dataStreamAutoShardingService.calculate(projectState, dataStream, indexWriteLoad);
+                    rolloverAutoSharding = dataStreamAutoShardingService.calculate(
+                        projectState,
+                        dataStream,
+                        indexStats.map(stats -> sumLoadMetrics(stats, IndexingStats.Stats::getWriteLoad)).orElse(null),
+                        indexStats.map(stats -> sumLoadMetrics(stats, IndexingStats.Stats::getRecentWriteLoad)).orElse(null),
+                        indexStats.map(stats -> sumLoadMetrics(stats, IndexingStats.Stats::getPeakWriteLoad)).orElse(null)
+                    );
```

Comment:
Thinking out loud: what if we moved the write load calculations in the `dataStreamAutoShardingService.calculate(...)` and just pass the `indexStats`? 

I think it fits the responsibility of the `DataStreamAutoShardingService.java` better and it can potentially allow us to do further improvements, if we deem that some write loads are not relevant.

What do you think?